src/algorithm/mixed_radix.rs

```diff
@@ -151,6 +161,47 @@ impl<T: FftNum> MixedRadix<T> {
         transpose::transpose(scratch, buffer, self.width, self.height);
     }
 
+    fn perform_fft_immut(
+        &self,
+        input: &[Complex<T>],
+        output: &mut [Complex<T>],
+        scratch_raw: &mut [Complex<T>],
+    ) {
+        // We require twice as much scratch here as perform_fft_out_of_place
+        // We have this psuedocode:
+        // ...
+        // fft(output, scratch) // FFT in place with scratch
+        // transpose(output, scratch) // Transpose output -> scratch
+        // fft(scratch, output) // FFT in place using `output` as scratch
+        // ...
+        // process_with_scratch can transpose the output into the input variable, saving scratch for just fft scratch
+        // Since we can't use the input variable, allocate twice as much scratch as process_with_scratch,
+        // and split them into two scratches we can use
+        let (scratch, scratch2) = scratch_raw.split_at_mut(scratch_raw.len() / 2);
```

Comment:
Same revised scratch-tracking algorithm as I wrote for good-thomas algorithm. Pass the whole thing to the height FFT, then split_at_mut(self.len()), transpose into the left scratch, and pass the right scratch to the width FFT.